src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecProc.java

```diff
@@ -86,7 +92,24 @@ public int join() throws IOException, InterruptedException {
             LOGGER.log(Level.FINEST, "Waiting for websocket to close on command finish ({0})", finished);
             finished.await();
             LOGGER.log(Level.FINEST, "Command is finished ({0})", finished);
-            return watch.exitCode().join();
+
+            CompletableFuture<Integer> exitCodeFuture = watch.exitCode();
+            Integer exitCode = exitCodeFuture.get();
+
+            if (exitCode == null) {
+                LOGGER.log(Level.FINEST, "The container exec watch was closed before it could obtain an exit code from the process.");
+                printStream.print("The container exec watch was closed before it could obtain an exit code from the process.");
+                return -1;
+            }
+            return exitCode;
+        } catch (ExecutionException e) {
+            LOGGER.log(Level.FINEST, "ExecutionException occurred while waiting for exit code", e.getCause());
+            printStream.printf("ExecutionException occurred while waiting for exit code: %s%n", e.getCause());
+            return -1;
+        } catch (Exception e) {
+            LOGGER.log(Level.FINEST, "Exception occurred while waiting for exit code", e.getCause());
+            printStream.printf("Exception occurred while waiting for exit code: %s: %s%n", e.getClass(), e.getCause());
```

Comment:
```suggestion
            LOGGER.log(Level.FINEST, "Exception occurred while waiting for exit code", e);
            printStream.printf("Exception occurred while waiting for exit code: %s%n", e);
```
or use `Functions.printStackTrace`